libs/testing/fixtures/src/lib/nationalId.ts

```diff
@@ -10,40 +10,33 @@ export type NationalIdType =
   | 'residentChild'
   | 'temporaryResident'
 
-const YEAR = 1000 * 60 * 60 * 24 * 365
-
 export const createNationalId = (type?: NationalIdType): string => {
   let date: Date
-  let variant: number
   switch (type) {
     case 'company':
-      date = new Date(Date.now() - faker.datatype.number(100 * YEAR))
+      date = faker.date.past({ years: 100 })
       return generateCompany(date)
     case 'person':
-      variant = faker.datatype.number(8)
-      return variant === 0
+      return faker.datatype.boolean(0.1111)
         ? createNationalId('temporaryResident')
         : createNationalId('resident')
     case 'resident':
-      variant = faker.datatype.number(4)
-      return variant === 0
+      return faker.datatype.boolean(0.2)
         ? createNationalId('residentChild')
         : createNationalId('residentAdult')
     case 'residentAdult':
-      date = new Date(
-        Date.now() - faker.datatype.number({ min: 18 * YEAR, max: 100 * YEAR }),
-      )
+      date = faker.date.birthdate({ mode: 'age', min: 18, max: 100 })
       return generatePerson(date)
     case 'residentChild':
-      date = new Date(Date.now() - faker.datatype.number(18 * YEAR))
+      date = faker.date.birthdate({ mode: 'age', min: 1, max: 17 })
       return generatePerson(date)
     case 'temporaryResident':
-      return faker.helpers.replaceSymbolWithNumber(
-        faker.helpers.regexpStyleStringParse('[8-9]#########'),
+      return (
+        faker.number.int({ min: 8, max: 9 }).toString() +
+        faker.string.numeric(9)
       )
```

Comment:
There was no runtime validation before as well, thats why I keeped it that way. The maintainers should decide whether this is required or not.

Reply:
Since we only choose between two numbers here it might be easier to understand if we use `faker.datatype.boolean() ? '8' : '9'` or `faker.helpers.arrayElement(['8', '9'])`. But this could be done in later PRs.